pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -249,8 +249,8 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
             filepath: write-target file's path
         """
         # dump states as a checkpoint dictionary object
+        checkpoint = self.on_save(checkpoint)
```

Comment:
let's split this out as a separate PR to land right away?

Reply:
PR made here: https://github.com/PyTorchLightning/pytorch-lightning/pull/7134